### ISSUE TITLE
Site Editor: fix the 'back to Dashboard' links labelling

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#49657]
### Description

In the Site Editor, there are two links to go back to the Dashboard. They're visually presented as 'icon buttons'.

The first one is the WP / site logo. It's labelled `aria-label="Go back to the dashboard"`.
The second one is the 'back' icon button. It's labelled `aria-label="Navigate to the Dashboard"`.

Also:
- The WP / sito logo doesn't have a tooltip. Icon buttons _must_ visually expose their accessible name in some way. At the vry least, with a tooltip.
- The back icon button tooltip text is just `Dashboard`, which mismatches the actual accessible name.

<img width="901" alt="go back links" src="https://user-images.githubusercontent.com/1682452/230621057-436d66ad-5ada-4f37-858c-81a2d8069103.png">

<img width="361" alt="tooltip" src="https://user-images.githubusercontent.com/1682452/230621095-b26aaab6-b86d-472f-ad46-a10b31f071d3.png">

This is less than ideal from an accessibility perspective.
- Controls that do the same thing should be labelled consistently.
- A control's tooltip is meant to visually expose its accessible name. Tooltip text and aria-label must match.

Also, ideally the visuals of an icon button should clearly suggest its actual accessible name. Users should be able to guess the actual accessible name by the icon. This is particularly important for speech recognition software users. As a user:
- I do see an icon that looks like a left arrow. That mimis the well-known convention of the browsers back button. It suggest me the idea of 'back', 'go back;.
- Based on that, I'd try a voice command like `Click 'back'`. 
- That voice command would fail because the actual accessible name is `Navigate to the Dashboard`.

### Step-by-step reproduction instructions

- Go to the Site editor in 'browse mode'.
- Inspect the source of the WP / site logo at the top left.
- Observe the lionk aria-label is `Go back to the dashboard`.
- Observe the link doesn't show a tooltip on hover or focus.
- Inspect the source of the back icon button.
- Observe the link aria-label is `Navigate to the Dashboard`.
- Observe the link tooltip is `Dashboard`.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by annezazu** [comment]
Going to add this to 6.2.1's board for consideration!
